s3_bucket.go

```diff
@@ -1397,6 +1403,7 @@ type PreSignRequestParams struct {
 	AwsSecret       string
 	AwsSessionToken string
 	Region          string
+	Expiry          time.Duration
```

Comment:
I'd think to call it `SignatureExpiryWindow` or something, just to make it clearer what it's for.